File: java/scf-apollo/src/main/java/org/mydotey/scf/apollo/ApolloConfigurationSourceConfig.java
package org.mydotey.scf.apollo;

import java.util.Objects;

import org.mydotey.scf.DefaultConfigurationSourceConfig;

import com.ctrip.framework.apollo.Config;

/**
 * @author koqizhao
 *
 * Jul 24, 2018
 */
public class ApolloConfigurationSourceConfig extends DefaultConfigurationSourceConfig {

    private Config _apolloConfig;

    protected ApolloConfigurationSourceConfig() {

    }

    public Config getApolloConfig() {
        return _apolloConfig;
    }

    public static class Builder extends DefaultConfigurationSourceConfig.DefaultAbstractBuilder<Builder> {

        @Override
        protected DefaultConfigurationSourceConfig newConfig() {
            return new ApolloConfigurationSourceConfig();
        }

        @Override
        protected ApolloConfigurationSourceConfig getConfig() {
            return (ApolloConfigurationSourceConfig) super.getConfig();
        }

        public Builder setApolloConfig(Config config) {
            getConfig()._apolloConfig = config;
            return this;
        }

        @Override
        public ApolloConfigurationSourceConfig build() {
            Objects.requireNonNull(getConfig()._apolloConfig, "apolloConfig is null");

            return (ApolloConfigurationSourceConfig) super.build();
        }

    }

}

File: dotnet/src/MyDotey.SCF.Apollo/ApolloConfigurationSourceConfig.cs
using System;

using Com.Ctrip.Framework.Apollo;

namespace MyDotey.SCF.Apollo
{
    /**
     * @author koqizhao
     *
     * Jul 24, 2018
     */
    public class ApolloConfigurationSourceConfig : DefaultConfigurationSourceConfig
    {
        private IConfig _apolloConfig;

        protected ApolloConfigurationSourceConfig()
        {

        }

        public virtual IConfig ApolloConfig { get { return _apolloConfig; } }

        public new class Builder
                : DefaultConfigurationSourceConfig.DefaultAbstractBuilder<Builder, ApolloConfigurationSourceConfig>
        {
            protected override ApolloConfigurationSourceConfig NewConfig()
            {
                return new ApolloConfigurationSourceConfig();
            }

            public Builder SetApolloConfig(IConfig config)
            {
                Config._apolloConfig = config;
                return this;
            }

            public override ApolloConfigurationSourceConfig Build()
            {
                if (Config._apolloConfig == null)
                    throw new ArgumentNullException("apolloConfig is null");

                return base.Build();
            }
        }
    }
}